### ISSUE TITLE
runtime: support debugCall on arm64

### ISSUE THREAD

**Issue by aarzilli** [golang/go#50614]
debugCall is currently only supported on amd64, with the release of arm64 based macs it would be nice if a version of it also existed on arm64 and the feature has been requested on Delve's issue tracker: https://github.com/go-delve/delve/issues/2277.

For example my message pointing out [go1.18 will not support it](https://github.com/go-delve/delve/issues/2277#issuecomment-979347623) received 25 sadfaces.

cc @heschi

**Comment by erifan** [comment]
I'll take a look, assign it to me for now, if anyone makes a patch before me, feel free to submit it. Thanks.

**Comment by mjaniko** [comment]
The issue is still open

**Comment by amjd** [comment]
Is this definitely not going to be included in go1.18, or is there a small chance it might?

**Comment by ianlancetaylor** [comment]
I don't see any implementation yet, so it will definitely not be in Go 1.18.  Sorry.

**Comment by amjd** [comment]
No problem, thanks for confirming @ianlancetaylor. Hope this gets implemented and is released with Go 1.19.

**Comment by erifan** [comment]
I have sent the PR, hope it will be merged into go1.19.

**Comment by AlekSi** [comment]
That CL is only for linux/arm64, not for darwin/arm64, right?

**Comment by erifan** [comment]
> 

Yes, X86 also currently supports Linux only.

**Comment by amjd** [comment]
Anyone planning to work on it for darwin/arm64? // @ianlancetaylor

**Comment by ianlancetaylor** [comment]
I'm not too familiar with `debugCall`, but what makes it Linux only?

**Comment by aarzilli** [comment]
AFAIK nothing.

**Comment by ianlancetaylor** [comment]
Let me rephrase: as far as I can see it should work on macOS/amd64 today.  Is that not the case?  If so, why doesn't it work?  Thanks.

**Comment by aarzilli** [comment]
I should have been less curt. As far as I know it work on macOS, today, as it is. It is not true that it is linux-only.

**Comment by ianlancetaylor** [comment]
Great, thanks.

**Comment by erifan** [comment]
Sorry I just stared at the test, the test only supports Linux, but the implementation should not be limited to the system.

**Comment by chenzhiguo** [comment]
Does Apple M1 (Pro, Max) will be supported?

**Comment by ncomet** [comment]
> Does Apple M1 (Pro, Max) will be supported?

From the discussions above, yes. It should be covered.

**Comment by ms140569** [comment]
This issue is still not solved on the Latest Mac OS X version:

Mac OSX: 12.3.1 (21E258)

Darwin FVFGT4PMQ05N 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:47:26 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T8101 arm64

go version go1.18.1 darwin/arm64

Delve Debugger
Version: 1.8.3
Build: $Id: f92bb46b82b3b92d79ce59c4b55eeefbdd8d040c $

**Comment by aarzilli** [comment]
@ms140569 this issue is about implementing the runtime's side of call injection, which did happen. It will not work until delve's side is also implemented.

**Comment by amjd** [comment]
Even if this is implemented, the changes won't be live until go1.19, right? Or is there a chance these could be included in a minor version?

**Comment by tinyfluffs** [comment]
@gopherbot please open backports for Go 1.18

**Comment by aclements** [comment]
@erifan , would you mind preparing a backport CL to release-branch.go.1.18 to resolve #52699?

**Comment by erifan** [comment]
Sure, I'd happy to.
